src/app/dashboard/page.jsx

```diff
@@ -52,6 +53,18 @@ export default function Page() {
         const fullUserActivity = await getUserActivity(user.uid);
         setUserActivity(fullUserActivity);
         setLoading(false);
+        // Check if error_message exists in params
+      if (errorSocial) {
+        setNotification({
+          type: "error",
+          message: errorSocial, // Set the error message from the URL param
+          show: true,
+        });
+        const newParams = new URLSearchParams(params.toString());
+        newParams.delete("error_message");
```

Comment:
depending on the "error" you need to either show a custom message or a generic message
if the error=access_denied then show a message such as "Can't connect, rejected by user."
and when the error type is something else (different than access_denied, display generic message such as "Unable to connect at the moment"